File: check-outbound-data.js
/**
 * 检查出库单数据
 */

const Database = require('better-sqlite3');
const path = require('path');

const DB_PATH = path.join(__dirname, 'data', 'database.sqlite');

console.log('🔍 检查数据库文件:', DB_PATH);

try {
  const db = new Database(DB_PATH, { readonly: true });
  
  console.log('\n📋 检查表是否存在...');
  const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table' AND name IN ('sales_outbound', 'sales_outbound_items', 'customers', 'warehouses', 'products')").all();
  console.log('找到的表:', tables.map(t => t.name));
  
  console.log('\n📋 检查销售出库主表数据...');
  const outbounds = db.prepare('SELECT * FROM sales_outbound').all();
  console.log(`出库单数量：${outbounds.length}`);
  
  if (outbounds.length > 0) {
    const firstOutbound = outbounds[0];
    console.log('\n第一条出库单数据:');
    console.log('ID:', firstOutbound.id);
    console.log('凭证号:', firstOutbound.outbound_no);
    console.log('客户 ID:', firstOutbound.customer_id);
    console.log('仓库 ID:', firstOutbound.warehouse_id);
    console.log('经办人:', firstOutbound.handler_name);
    console.log('操作员:', firstOutbound.created_by);
    console.log('总金额:', firstOutbound.total_amount);
    
    console.log('\n📋 检查客户表...');
    const customers = db.prepare('SELECT * FROM customers').all();
    console.log(`客户数量：${customers.length}`);
    if (customers.length > 0) {
      console.log('第一个客户:', customers[0]);
    }
    
    console.log('\n📋 检查仓库表...');
    const warehouses = db.prepare('SELECT * FROM warehouses').all();
    console.log(`仓库数量：${warehouses.length}`);
    if (warehouses.length > 0) {
      console.log('第一个仓库:', warehouses[0]);
    }
    
    console.log('\n📋 检查出库单明细...');
    const items = db.prepare('SELECT * FROM sales_outbound_items WHERE outbound_id = ?').all(firstOutbound.id);
    console.log(`明细数量：${items.length}`);
    
    if (items.length > 0) {
      console.log('\n第一条明细数据:');
      const firstItem = items[0];
      console.log('商品 ID:', firstItem.product_id);
      console.log('商品名称:', firstItem.product_name);
      console.log('规格:', firstItem.specification);
      console.log('单位:', firstItem.unit);
      console.log('数量:', firstItem.quantity);
      console.log('单价 (不含税):', firstItem.unit_price);
      console.log('单价 (含税):', firstItem.sale_price);
      console.log('税率:', firstItem.tax_rate);
      console.log('税额:', firstItem.tax_amount);
      console.log('总额:', firstItem.total_amount);
      console.log('成本价:', firstItem.cost_price);
      
      console.log('\n📋 检查商品表...');
      const products = db.prepare('SELECT * FROM products').all();
      console.log(`商品数量：${products.length}`);
      if (products.length > 0) {
        const product = products.find(p => p.id === firstItem.product_id);
        console.log('对应的商品:', product);
      }
    }
  } else {
    console.log('❌ 没有找到出库单数据！');
  }
  
  db.close();
  console.log('\n✅ 检查完成');
} catch (error) {
  console.error('❌ 检查失败:', error.message);
  console.error(error.stack);
}

File: check-outbound-data.py
import sqlite3

# 数据库路径
db_path = r'C:\Users\Administrator\AppData\Roaming\Electron\inventory.db'

print(f"🔍 检查出库单数据...")
print(f"数据库路径：{db_path}")

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # 查询所有出库单
    print("\n📋 查询所有出库单...")
    cursor.execute("""
        SELECT id, outbound_no, outbound_date, total_amount, status, created_at 
        FROM sales_outbound 
        ORDER BY created_at DESC 
        LIMIT 5
    """)
    outbounds = cursor.fetchall()
    
    if not outbounds:
        print("❌ 没有找到出库单记录")
    else:
        print(f"\n✅ 找到 {len(outbounds)} 条出库单记录:")
        for outbound in outbounds:
            print(f"\n  出库单 #{outbound[0]}: {outbound[1]}")
            print(f"    日期：{outbound[2]}")
            print(f"    总金额：{outbound[3]}")
            print(f"    状态：{outbound[4]}")
            print(f"    创建时间：{outbound[5]}")
            
            # 查询明细
            cursor.execute("""
                SELECT product_id, product_name, specification, unit, 
                       quantity, unit_price, sale_price, tax_rate, tax_amount, total_amount, cost_price, remark
                FROM sales_outbound_items 
                WHERE outbound_id = ?
            """, (outbound[0],))
            
            items = cursor.fetchall()
            print(f"    明细数量：{len(items)}")
            
            for i, item in enumerate(items, 1):
                print(f"\n      明细 #{i}:")
                print(f"        产品 ID: {item[0]}")
                print(f"        产品名称：{item[1] or '(空)'}")
                print(f"        规格：{item[2] or '(空)'}")
                print(f"        单位：{item[3] or '(空)'}")
                print(f"        数量：{item[4]}")
                print(f"        进货价：{item[5] or '(空)'}")
                print(f"        售价：{item[6] or '(空)'}")
                print(f"        税率：{item[7] or '(空)'}")
                print(f"        税额：{item[8] or '(空)'}")
                print(f"        总金额：{item[9] or '(空)'}")
                print(f"        成本价：{item[10] or '(空)'}")
                print(f"        备注：{item[11] or '(空)'}")
    
    conn.close()
    print("\n✅ 检查完成")
    
except Exception as e:
    print(f"❌ 检查失败：{e}")
    import traceback
    traceback.print_exc()
